code/weapon/weapons.cpp

```diff
@@ -1261,6 +1261,12 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 	if (optional_string("@Laser Glow Tail Scale:")) {
 		stuff_float(&wip->laser_glow_tail_scale);
 	}
+  
+	float mpsbuffer;
+	if (parse_optional_float_into("@Min Pixel Size:", &mpsbuffer))
+		wip->min_pixel_size = mpsbuffer;
+	else
+		wip->min_pixel_size = Min_pixel_size_laser;
```

Comment:
This should not be handled using the else case, but rather be set as default during the reset method of the weapon_info struct.

Reply:
And if you do so, you also don't need the mpsbuffer variable. You can either do it using the if-optionalstring-then-stuff method, or parse-optional-float-into directly

Reply:
Made the requested changes.